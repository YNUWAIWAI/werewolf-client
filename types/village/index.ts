/* eslint no-unused-vars: 0, @typescript-eslint/no-unused-vars: 0 */
export * from './payload'

export const enum BaseContext {
  Base = 'https://werewolf.world/village/context/0.3/base.jsonld',
  Board = 'https://werewolf.world/village/context/0.3/board.jsonld',
  Character = 'https://werewolf.world/village/context/0.3/character.jsonld',
  Chat = 'https://werewolf.world/village/context/0.3/chat.jsonld',
  Error = 'https://werewolf.world/village/context/0.3/error.jsonld',
  FlavorText = 'https://werewolf.world/village/context/0.3/flavorText.jsonld',
  Role = 'https://werewolf.world/village/context/0.3/role.jsonld',
  Scroll = 'https://werewolf.world/village/context/0.3/scroll.jsonld',
  Star = 'https://werewolf.world/village/context/0.3/star.jsonld',
  Vote = 'https://werewolf.world/village/context/0.3/vote.jsonld',
  VotingResult = 'https://werewolf.world/village/context/0.3/votingResult.jsonld'
}
export const enum BoardState {
  CIRCLE = 'O',
  CROSS = 'X',
  FILL = 'fill',
  QUESTION = '?',
  TRIANGLE = 'Δ'
}
export const enum Channel {
  anonymousAudience = 'anonymousAudience',
  grave = 'grave',
  hunter = 'hunter',
  master = 'master',
  onymousAudience = 'onymousAudience',
  private = 'private',
  public = 'public',
  seer = 'seer',
  werewolf = 'werewolf'
}
export type CharacterId = string
export const enum CharacterStatus {
  alive = 'alive',
  dead = 'dead',
  deathByAttack = 'death by attack',
  deathByExecution = 'death by execution',
  deathByFear = 'death by fear',
  unnaturalDeath = 'unnatural death'
}
export const enum ChatChannel {
  grave = 'grave',
  master = 'master',
  private = 'private',
  public = 'public',
  werewolf = 'werewolf'
}
export type ChatId = string
export const enum ChatItemType {
  delimeter = 'delimeter',
  item = 'item'
}
export const enum Context {
  Avatar = 'https://werewolf.world/village/context/0.3/avatar.jsonld',
  Base = 'https://werewolf.world/village/context/0.3/base.jsonld',
  Board = 'https://werewolf.world/village/context/0.3/board.jsonld',
  BoardResult = 'https://werewolf.world/village/context/0.3/boardResult.jsonld',
  Character = 'https://werewolf.world/village/context/0.3/character.jsonld',
  Chat = 'https://werewolf.world/village/context/0.3/chat.jsonld',
  ChatSettings = 'https://werewolf.world/village/context/0.3/chatSettings.jsonld',
  Error = 'https://werewolf.world/village/context/0.3/error.jsonld',
  FlavorText = 'https://werewolf.world/village/context/0.3/flavorText.jsonld',
  Role = 'https://werewolf.world/village/context/0.3/role.jsonld',
  Scroll = 'https://werewolf.world/village/context/0.3/scroll.jsonld',
  Star = 'https://werewolf.world/village/context/0.3/star.jsonld',
  Village = 'https://werewolf.world/village/context/0.3/village.jsonld',
  Vote = 'https://werewolf.world/village/context/0.3/vote.jsonld',
  VotingResult = 'https://werewolf.world/village/context/0.3/votingResult.jsonld'
}
export const enum Directionality {
  serverToClient = 'server to client',
  clientToServer = 'client to server'
}
export const enum Gender {
  female = 'female',
  male = 'male'
}
export const enum InputChannel {
  grave = 'grave',
  postMortem = 'post mortem',
  private = 'private',
  public = 'public',
  werewolf = 'werewolf'
}
export const enum Language {
  en = 'en',
  fr = 'fr',
  it = 'it',
  ja = 'ja'
}
export interface LanguageMap {
  [Language.en]: string
  [Language.fr]?: string
  [Language.it]?: string
  [Language.ja]?: string
}
export type Payload$AgentId = number
export type Payload$ChatId = number
export const enum Phase {
  flavorText = 'flavor text',
  morning = 'morning',
  night = 'night',
  noon = 'noon',
  postMortem = 'post mortem',
  result = 'result'
}
export const enum Polarity {
  circle = 'O',
  cross = 'X',
  negative = 'negative',
  positive = 'positive',
  question = '?',
  triangle = 'Δ'
}
export const enum Result {
  lose = 'lose',
  win = 'win'
}
export const enum RoleId {
  hunter = 'hunter',
  madman = 'madman',
  mason = 'mason',
  master = 'master',
  medium = 'medium',
  seer = 'seer',
  villager = 'villager',
  werehamster = 'werehamster',
  werewolf = 'werewolf'
}
export const enum Severity {
  error = 'error',
  warning = 'warning'
}
export const enum SummaryType {
  audience = 'audience',
  character = 'character'
}
export const enum Team {
  villager = 'villager',
  werehamster = 'werehamster',
  werewolf = 'werewolf'
}
export type Token = string
export type VillageId = number
