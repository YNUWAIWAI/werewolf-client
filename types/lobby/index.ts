import {Avatar as AvatarScehma} from '../village/schema'
import {BuildVillage} from './schema'

export {
  Payload,
  PayloadType,
  Payload$AdvancedSearch,
  Payload$BuildVillage,
  Payload$Avatar,
  Payload$ChangeLanguage,
  Payload$ChangeUserEmail,
  Payload$ChangeUserName,
  Payload$ChangeUserPassword,
  Payload$EnterLobby,
  Payload$GetAvatar,
  Payload$GetSettings,
  Payload$IdSearch,
  Payload$KickOutPlayer,
  Payload$LeaveWaitingPage,
  Payload$Lobby,
  Payload$Ping,
  Payload$Play,
  Payload$Played,
  Payload$Pong,
  Payload$Ready,
  Payload$SearchResult,
  Payload$SelectVillage,
  Payload$Settings,
  Payload$WaitingPage
} from './payload'
export {Language} from '../village'
export const enum Avatar {
  fixed = 'fixed',
  random = 'random',
  unspecified = 'unspecified'
}
export const enum Member {
  A = 'A',
  B = 'B',
  C = 'C'
}
export const enum LobbyType {
  anonymousAudience = 'anonymous audience',
  onymousAudience = 'onymous audience',
  human = 'human player',
  robot = 'robot player'
}
export const enum PingStatus {
  danger = 'danger',
  safe = 'safe',
  warning = 'warning'
}
export const enum RoleId {
  hunter = 'hunter',
  madman = 'madman',
  mason = 'mason',
  medium = 'medium',
  seer = 'seer',
  villager = 'villager',
  werehamster = 'werehamster',
  werewolf = 'werewolf'
}
export type Village = Omit<BuildVillage, 'token' | 'type'>
export interface Cast {
  [numberOfPlayers: string]: {
    [member in Member]: BuildVillage['roleSetting']
  }
}
export type Token = AvatarScehma['token']
export interface WaitingPlayer {
  avatarImage: string
  isAnonymous: boolean
  isHost: boolean
  isMe: boolean
  name: string
  token: Token
}
export interface PingResult {
  ping: string
  status: PingStatus
  token: AvatarScehma['token']
}
export const enum TestStatus {
  notPassed = 'not passed',
  passed = 'passed'
}
export const enum Automation {
  full = 'full',
  semi = 'semi'
}
export const enum Authorized {
  no = 'no',
  yes = 'yes',
  waitForAcceptance = 'wait for acceptance',
}
export const enum AvatarStatus {
  awaitingAuthorization = 'awaiting authorization',
  awaitingCommunicationTest = 'awaiting communication test',
  connected = 'connected',
  runningInTheBackground = 'running in the background',
  runningInTheForeground = 'running in the foreground'
}
