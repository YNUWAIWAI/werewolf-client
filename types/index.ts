import * as lobby from './lobby'
import * as village from './village'

export {
  lobby,
  village
}
