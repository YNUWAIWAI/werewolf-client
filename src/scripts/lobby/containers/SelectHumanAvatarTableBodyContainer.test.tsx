import * as ActionTypes from '../constants/ActionTypes'
import * as React from 'react'
import {IntlProviderContainer} from './IntlProviderContainer'
import {Provider} from 'react-redux'
import {SelectHumanAvatarTableBody} from '../components/molecules/SelectHumanAvatarTableBody'
import {SelectHumanAvatarTableBodyContainer} from './SelectHumanAvatarTableBodyContainer'
import fakeStore from './fakeStore'
import {lobby} from '../types'
import {mount} from 'enzyme'

describe('<SelectHumanAvatarTableBodyContainer />', () => {
  test('render', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <SelectHumanAvatarTableBodyContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('handleAvatarImageClick', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <SelectHumanAvatarTableBodyContainer />
        </IntlProviderContainer>
      </Provider>
    )
    const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
    const image = 'image'

    wrapper.find(SelectHumanAvatarTableBody).props().handleAvatarImageClick(token)(image)
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      image,
      token,
      type: ActionTypes.SelectHumanAvatar.SHOW_AVATAR_IMAGE_SELECT
    })
  })
  test('handleAvatarLanguageChange', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <SelectHumanAvatarTableBodyContainer />
        </IntlProviderContainer>
      </Provider>
    )
    const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
    const language = lobby.Language.en

    wrapper.find(SelectHumanAvatarTableBody).props().handleAvatarLanguageChange(token)(true)(language)
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      language,
      token,
      type: ActionTypes.SelectHumanAvatar.CHANGE_AVATAR_LANGUAGE
    })
  })
  describe('handleAvatarNameChange', () => {
    test('valid', () => {
      const store = fakeStore()
      const dispatch = jest.fn()

      store.dispatch = dispatch
      const wrapper = mount(
        <Provider store={store} >
          <IntlProviderContainer>
            <SelectHumanAvatarTableBodyContainer />
          </IntlProviderContainer>
        </Provider>
      )
      const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
      const name = 'avatarName'

      wrapper.find(SelectHumanAvatarTableBody).props().handleAvatarNameChange(token)(true)(name)
      expect(dispatch).toHaveBeenCalledTimes(1)
      expect(dispatch).toHaveBeenCalledWith({
        name,
        token,
        type: ActionTypes.SelectHumanAvatar.CHANGE_AVATAR_NAME
      })
    })
    test('invalid', () => {
      const store = fakeStore()
      const dispatch = jest.fn()

      store.dispatch = dispatch
      const wrapper = mount(
        <Provider store={store} >
          <IntlProviderContainer>
            <SelectHumanAvatarTableBodyContainer />
          </IntlProviderContainer>
        </Provider>
      )
      const token = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'
      const name = 'avatar'

      wrapper.find(SelectHumanAvatarTableBody).props().handleAvatarNameChange(token)(false)(name)
      expect(dispatch).toHaveBeenCalledTimes(0)
    })
  })
  test('handleHoverAvatar', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <SelectHumanAvatarTableBodyContainer />
        </IntlProviderContainer>
      </Provider>
    )
    const id = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'

    wrapper.find(SelectHumanAvatarTableBody).props().handleHoverAvatar(id)()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      id,
      type: ActionTypes.SelectHumanAvatar.HOVER_AVATAR
    })
  })
  test('handleSelectAvatar', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <SelectHumanAvatarTableBodyContainer />
        </IntlProviderContainer>
      </Provider>
    )
    const id = '3F2504E0-4F89-11D3-9A0C-0305E82C3300'

    wrapper.find(SelectHumanAvatarTableBody).props().handleSelectAvatar(id)()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      id,
      type: ActionTypes.SelectHumanAvatar.CHANGE_CHECKBOX
    })
  })
})
