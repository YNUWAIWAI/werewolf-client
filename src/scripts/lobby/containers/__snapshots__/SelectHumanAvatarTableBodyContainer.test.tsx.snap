// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectHumanAvatarTableBodyContainer /> render 1`] = `""`;
